subprojects/snapshots/src/main/java/org/gradle/internal/vfs/VfsRoot.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.vfs;
+
+import org.gradle.internal.snapshot.MetadataSnapshot;
+import org.gradle.internal.snapshot.ReadOnlyVfsRoot;
+
+import javax.annotation.CheckReturnValue;
+
+public interface VfsRoot extends ReadOnlyVfsRoot {
```

Comment:
I think this should be `VfsRoot` and `MutableVfsRoot`. But I'm not convinced it's good to have a mutable version like this.